numpy/core/src/multiarray/common.h

```diff
@@ -181,6 +182,18 @@ check_and_adjust_axis(int *axis, int ndim)
 {
     return check_and_adjust_axis_msg(axis, ndim, Py_None);
 }
+static NPY_INLINE int
+check_and_adjust_axis_cmsg(int *axis, int ndim, char const *cmsg)
+{
+    int ret;
+    PyObject *msg = PyUString_FromString(cmsg);
```

Comment:
Here, now you are adding a slow function that is only needed if an error is raised to what should otherwise be a super-fast call. I think you are better off adding a flag or some other argument parsing to the main routine so that it knows whether the prefix passed in is a C or Python string (maybe just allow passing in both, with one expected to be set to NULL?).

Reply:
Should I just construct the `"axis1"` and `"axis2"` strings statically at the call site?